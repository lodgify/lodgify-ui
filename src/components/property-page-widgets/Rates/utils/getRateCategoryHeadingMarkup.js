import React, { Fragment } from 'react';
import { isNil } from 'lodash';

import { Icon, ICON_NAMES } from 'elements/Icon';
import { Paragraph } from 'typography/Paragraph';

import { getCostPerExtraGuestString } from './getCostPerExtraGuestString';

/**
 * @param {Object} rateCategory
 * @param {string} rateCategory.costPerExtraGuest
 * @param {string} rateCategory.dateRange
 * @param {string} rateCategory.name
 * @param {string} rateCategory.numberOfGuests
 * @param {string} costPerExtraGuestLabel
 * @return {Object}
 */
export const getRateCategoryHeadingMarkup = (
  {
    /* eslint-disable react/prop-types */
    costPerExtraGuest,
    dateRange,
    name,
    numberOfGuests,
    /* eslint-enable react/prop-types */
  },
  costPerExtraGuestLabel
) => (
  <div>
    <Paragraph weight="heavy">{name}</Paragraph>
    <Paragraph weight="light">
      {dateRange}
      <br />
      {!isNil(numberOfGuests) && (
        <Fragment>
          <Icon name={ICON_NAMES.GUESTS} />
          {numberOfGuests}
          <br />
        </Fragment>
      )}
      {!isNil(costPerExtraGuest) &&
        getCostPerExtraGuestString(costPerExtraGuest, costPerExtraGuestLabel)}
    </Paragraph>
  </div>
);
