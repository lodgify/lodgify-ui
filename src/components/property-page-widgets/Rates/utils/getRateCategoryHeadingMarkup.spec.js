import { shallow } from 'enzyme';

import { getRateCategoryHeadingMarkup } from './getRateCategoryHeadingMarkup';

const costPerExtraGuestLabel = 'laaaabel';

const getRateHeading = rateCategory =>
  shallow(getRateCategoryHeadingMarkup(rateCategory, costPerExtraGuestLabel));

describe('getRateCategoryHeadingMarkup', () => {
  describe('if all values are defined', () => {
    it('should return the right markup', () => {
      const actual = getRateHeading({
        name: 'Mid Season',
        dateRange: '01/05/2018 - 01/08/2018',
        numberOfGuests: '2',
        costPerExtraGuest: '5€',
      });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `rateCategory.numberOfGuests` is `null` or `undefined`', () => {
    it('should return the right markup', () => {
      const testCases = [
        {
          name: 'Mid Season',
          dateRange: '01/05/2018 - 01/08/2018',
          numberOfGuests: null,
          costPerExtraGuest: '5€',
        },
        {
          name: 'High Season',
          dateRange: '01/08/2018 - 01/10/2018',
          costPerExtraGuest: '50€',
        },
      ];

      testCases.forEach(testCase => {
        const actual = getRateHeading(testCase);

        expect(actual).toMatchSnapshot();
      });
    });
  });

  describe('if `rateCategory.costPerExtraGuest` is `null` or `undefined`', () => {
    it('should return the right markup', () => {
      const testCases = [
        {
          name: 'Mid Season',
          dateRange: '01/05/2018 - 01/08/2018',
          numberOfGuests: '5',
          costPerExtraGuest: null,
        },
        {
          name: 'High Season',
          dateRange: '01/08/2018 - 01/10/2018',
          numberOfGuests: '19',
        },
      ];

      testCases.forEach(testCase => {
        const actual = getRateHeading(testCase);

        expect(actual).toMatchSnapshot();
      });
    });
  });
});
