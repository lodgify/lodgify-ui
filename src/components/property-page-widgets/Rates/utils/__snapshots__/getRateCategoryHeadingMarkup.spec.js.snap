// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRateCategoryHeadingMarkup if \`rateCategory.costPerExtraGuest\` is \`null\` or \`undefined\` should return the right markup 1`] = `
<div>
  <Paragraph
    size="medium"
    weight="heavy"
  >
    Mid Season
  </Paragraph>
  <Paragraph
    size="medium"
    weight="light"
  >
    01/05/2018 - 01/08/2018
    <br />
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={false}
      isLabelLeft={false}
      labelText={null}
      labelWeight={null}
      name="guests"
      path={null}
      size={null}
    />
    5
    <br />
  </Paragraph>
</div>
`;

exports[`getRateCategoryHeadingMarkup if \`rateCategory.costPerExtraGuest\` is \`null\` or \`undefined\` should return the right markup 2`] = `
<div>
  <Paragraph
    size="medium"
    weight="heavy"
  >
    High Season
  </Paragraph>
  <Paragraph
    size="medium"
    weight="light"
  >
    01/08/2018 - 01/10/2018
    <br />
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={false}
      isLabelLeft={false}
      labelText={null}
      labelWeight={null}
      name="guests"
      path={null}
      size={null}
    />
    19
    <br />
  </Paragraph>
</div>
`;

exports[`getRateCategoryHeadingMarkup if \`rateCategory.numberOfGuests\` is \`null\` or \`undefined\` should return the right markup 1`] = `
<div>
  <Paragraph
    size="medium"
    weight="heavy"
  >
    Mid Season
  </Paragraph>
  <Paragraph
    size="medium"
    weight="light"
  >
    01/05/2018 - 01/08/2018
    <br />
    5€ / laaaabel
  </Paragraph>
</div>
`;

exports[`getRateCategoryHeadingMarkup if \`rateCategory.numberOfGuests\` is \`null\` or \`undefined\` should return the right markup 2`] = `
<div>
  <Paragraph
    size="medium"
    weight="heavy"
  >
    High Season
  </Paragraph>
  <Paragraph
    size="medium"
    weight="light"
  >
    01/08/2018 - 01/10/2018
    <br />
    50€ / laaaabel
  </Paragraph>
</div>
`;

exports[`getRateCategoryHeadingMarkup if all values are defined should return the right markup 1`] = `
<div>
  <Paragraph
    size="medium"
    weight="heavy"
  >
    Mid Season
  </Paragraph>
  <Paragraph
    size="medium"
    weight="light"
  >
    01/05/2018 - 01/08/2018
    <br />
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={false}
      isLabelLeft={false}
      labelText={null}
      labelWeight={null}
      name="guests"
      path={null}
      size={null}
    />
    2
    <br />
    5€ / laaaabel
  </Paragraph>
</div>
`;
